### ISSUE TITLE
Window size and maximize state is not persisted after a restart of the app

### ISSUE THREAD

**Issue by sebastianhaeni** [MarshallOfSound/Google-Play-Music-Desktop-Player-UNOFFICIAL-#783]
**OS:**
Windows 10 Pro x64

**Issue Descriptions:**
Window size and maximize state is not persisted. When starting the app it's not where I last closed it. And it's not maximized if I closed it in a maximized state. I have two monitors in my setup. Maybe that's causing the problem.

**Steps to Reproduce:**
Move window / change maximize. Close app completely (not just to tray). Start the app again.

**Comment by MarshallOfSound** [comment]
Please provide more information as this feature is implemented, the app certainly saves position and size and attempts to restore them

**Comment by zabri** [comment]
Hi! I am also running Windows 10 Pro x64 on a dual screen setup and I can confirm this issue.

Here is what I have tried to do with the app and noticed about its behaviour:
1) Maximizing the app on either screen causes the app to overlap to the other screen. It's not, however, the orange border that overlaps, just a black strip (see image 1).
2) When the app overlaps like this (see 1), minimizing it and maximizing it back removes the overlap (see image 2).
3) Maximizing the app on the primary screen and closing it results in an expected behaviour - the app restarts on the primary monitor in a maximized state. This also removes the overlap if it was still there (see 1 and 2).
4) Maximizing the app on the secondary screen and closing it results in a restart in the middle of the primary screen. Its size, however, corresponds with the size of the secondary screen.
5) Putting the app into the middle of the secondary screen and closing it results in an expected behaviour - the app restarts where it was.
6) Putting the app on any screen with the top left-hand corner of the app behind the edge of any screen and closing it results in a restart in the middle of the primary screen. The size corresponds to the size it was in before the close (see images 3 and 4).

Good luck in fixing it and thank you for creating the app. ;)
![Uploading 1.png…]()
![2](https://cloud.githubusercontent.com/assets/18167148/14142908/bf648680-f688-11e5-94c2-28c166eb08ec.png)
![Uploading 3.png…]()
![4](https://cloud.githubusercontent.com/assets/18167148/14142907/bf63eeb4-f688-11e5-9a2c-5cea6cbbd6d9.png)

**Comment by MarshallOfSound** [comment]
@zabri Thanks for the detailed report, I can replicate the slightly overhanging black bar (I think that bit is an upstream electron issue) but I still can't replicate the situation where the app forgets about the position / ignore the position.

I'll keep looking into it

**Comment by abosmet** [comment]
Came to report this same issue.
Running Windows 7 Ultimate x64 with the 3.1.0 release version (updated from previous release).

TL;DR: Maximized state does not appear to be saved and causes odd behavior on restore.

For my setup, the right monitor is set as the primary, and the left is set as secondary.
I have the taskbar placed on the right side of the secondary monitor.
Both monitors have a resolution of 1440 x 900 px.

I have identified the following factors as being important:

Whether the window is on the primary or secondary monitor. (Primary Monitor / Secondary Monitor)
Whether the window was at a specific size, or maximized. (Sized / Maximized)
Whether the window was minimized to the system tray or the program was closed and reopened. (Minimized to Tray and restored / Closed and reopened)

To start, I sized the window pretty arbitrarily, but I wanted it at a smaller size so it was obvious what was going on if there were any changes.
Original dimensions ended up being 974 x 658 px.
I tried to visually align the left edge of the GPMDP window with the left edge of the Reddit icon on my hotbar.

##### Case 1: Primary Monitor, Sized, Minimized to Tray and restored:
Size and position restored correctly.

![](http://i.imgur.com/MmL7vzY.png)

![](http://i.imgur.com/hnZC18b.png)

##### Case 2: Primary Monitor, Maximized, Minimized to Tray and restored:
Size and position restored to the size and position it had before being maximized.

![](http://i.imgur.com/iHd8tKy.png)

![](http://i.imgur.com/IsDjfLr.png)

##### Case 3: Primary Monitor, Sized, Closed and reopened:
Size and position restored correctly.

![](http://i.imgur.com/NoqWdC9.png)

![](http://i.imgur.com/yqRXjfa.png)

##### Case 4: Primary Monitor, Maximized, Closed and reopened:
This is where things start to get weird. The window was restored to the size it had when it was maximized (1440 x 900, the full screen space of my monitor), but the window is not in a maximized state. I am able to resize the window with the mouse by pointing to the side or corners of the window, which is not possible when the window is actually maximized. Clicking the maximize button at this point adjusts the appearance of the window slightly and also removes the ability to resize the window using the mouse. 

![](http://i.imgur.com/PAivkMK.png)

![](http://i.imgur.com/sEDSmBG.png)

Beyond this point, I resized the window to smaller dimensions again so it is more obvious as to what is going on. Again, I visually aligned the left side of the window with the left side of the Reddit icon in the hotbar. I moved Chrome to the secondary monitor to align this. New dimensions are 948 x 635 px.

##### Case 5: Secondary Monitor, Sized, Minimized to Tray and restored:
Size and position restored correctly.

![](http://i.imgur.com/tcLWGC9.png)

![](http://i.imgur.com/yv7besc.png)

##### Case 6: Secondary Monitor, Maximized, Minimized to Tray and restored:
Size and position restored to the size and position it had before being maximized.

![](http://i.imgur.com/Fv3uyLx.png)

![](http://i.imgur.com/H8XcDoB.png)

##### Case 7: Secondary Monitor, Sized, Closed and reopened:
Size and position restored correctly.

![](http://i.imgur.com/XCtaINz.png)

![](http://i.imgur.com/1RMag65.png)

##### Case 8: Secondary Monitor, Maximized, Closed and reopened:
This is where things get really weird. The size of the window was restored to slightly larger than it was while maximized on the secondary monitor (was 1378 x 900, now 1394 x 900). BUT the window has been positioned centered on the primary monitor. There is equal space left and right of the window since maximizing the window on the secondary monitor does not cover the full screen space as it does on the primary monitor (taskbar is there).

![](http://i.imgur.com/xRY97sE.png)

![](http://i.imgur.com/nFiL2Gb.png)

I'm not 100% familiar with the source, but I would assume that all of this should be taking place within /main/features/core/persistAppState.js

In there, I am seeing this function which appears be saving the size and position if the mini player is not open:
```javascript
const _save = () => {
  if (!mini) {
    Settings.set('position', mainWindow.getPosition());
    Settings.set('size', mainWindow.getSize());
  } else {
    Settings.set('mini-position', mainWindow.getPosition());
    Settings.set('mini-size', mainWindow.getSize());

    // Keep the mini-player square.
    const dimension = Math.max(...mainWindow.getSize());
    if (resizeTimer) clearTimeout(resizeTimer);
    resizeTimer = setTimeout(() => mainWindow.setSize(dimension, dimension), 100);
    Emitter.sendToGooglePlayMusic('set:zoom', dimension / 310);
  }
};
```

That is done on any change in size or movement of the window, but I am having a difficult time following what is happening on maximize:
```javascript
mainWindow.on('move', _save);
mainWindow.on('resize', _save);
mainWindow.on('maximize', (ev) => {
  if (mini) {
    mainWindow.unmaximize();
    ev.preventDefault();
    return false;
  }
});
```

Based on the way this is behaving, I am thinking there is another spot where the state is saved.

I didn't want to go overboard with images here, so I consolidated all of this information in an Imgur album as well, which also includes screenshots of the window sizing with dimensions: [http://imgur.com/a/7Rd0S](url)

I hope this is helpful. Please let me know if anything is unclear or if there are any further cases you can think to test it in. Great app so far, I'm really looking forward to following the development of it.

**Comment by sebastianhaeni** [comment]
There's not much that I can add that has been reported already.

But for reference this is what happens when I have the app maximized on my left screen, then close to tray and open in again.

![Before](http://i.imgur.com/UVfCOc8.jpg)
![After](http://i.imgur.com/T4gCVJz.jpg)

Using system window borders makes no difference in this case.

**Comment by MarshallOfSound** [comment]
Thanks so much for all the detailed information guys.  I'm gone through all the use cases outlined above and fixed them to perform as expected :+1:
